File: PRACTICA3/QytetetJava/src/modeloqytetet/PruebaQytetet.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package modeloqytetet;

import java.util.ArrayList;

/**
 *
 * @author Jesus Manuel Garcia Palma
 */
public class PruebaQytetet {
       private static Qytetet juego = Qytetet.getInstance();
    
    public static void main(String[] args) {
       //PruebaQytetet prueba = new PruebaQytetet();
       juego.getNombreJugadores();
       /*System.out.println("Tablero\n");
       System.out.println(juego.getTablero().toString());
       System.out.println("Sorpresas\n");
       System.out.println(juego.getMazo().toString());
       
       System.out.println(prueba.metodo1());
       System.out.println(prueba.metodo2());
       
       TipoSorpresa[] t =  TipoSorpresa.values();
       for(TipoSorpresa tipo: t)
       {
          System.out.println(prueba.metodo3(tipo)); 
       }
       */
       System.out.println(juego.toString(
       
       ));
    }
    
    private ArrayList<Sorpresa> metodo1() //sorpresas cuyo valor es mayor a 0
    {
        ArrayList<Sorpresa> array = new ArrayList<>();
        for(Sorpresa s: juego.getMazo())
        {
            if(s.getValor()>0) array.add(s);
        }
        
        return array;
    }
    
    private ArrayList<Sorpresa> metodo2() //Sorpresas de TipoSorpresa IRACASILLA
    {
        ArrayList<Sorpresa> array = new ArrayList<>();
        for(Sorpresa s: juego.getMazo())
        {
            if(s.getTipo() == TipoSorpresa.IRACASILLA) array.add(s);
        }
        
        return array;
    }
    
    private ArrayList<Sorpresa> metodo3(TipoSorpresa tipo) //Sorpresas del TipoSorpresa indicado como argumento
    {
        ArrayList<Sorpresa> array = new ArrayList<>();
        for(Sorpresa s: juego.getMazo())
        {
            if(s.getTipo()==tipo) array.add(s);
        }
        
        return array;
    }
    
}

File: PRACTICA2/QytetetRuby/lib/prueba_qytetet.rb
#encoding: utf-8
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.

require_relative "qytetet.rb"
require_relative "tablero.rb"

module ModeloQytetet
  class PruebaQytetet
    @@juego = Qytetet.instance
    
    def main
      nombres = Array.new
      nombres << "Jugador1"
      nombres << "Jugador2"
     # @@juego.inicializar_juego(nombres)
      puts "Tablero\n"
      puts(@@juego)
      #puts "Mazo\n"
      #puts(@@juego.mazo)
      #puts(metodo_1)
       #puts(metodo_2)
      #arr = Array.new(TipoSorpresa::constants)
      #arr.each do |tipos|
        #puts metodo_3(tipos)
       # puts metodo_3(TipoSorpresa.const_get(tipos))
      #end
      
    end
    
    def metodo_1
      array = Array.new
      puts "\nSorpresas con valor mayor que cero\n"
      @@juego.mazo.each do |maz|
        if maz.valor > 0
          array<< maz
        end
        
       end
       return array
    end
    
    def metodo_2
      array = Array.new
      puts "\nSorpresas de TipoSorpresa IRACASILLA"
      @@juego.mazo.each do |maz|
        if maz.tipo == TipoSorpresa::IRACASILLA
          array<< maz
        end
      end
      return array
    end
    
    def metodo_3(tip)
      puts "\nSorpresas del tipo especificado\n"
      array = Array.new
      @@juego.mazo.each do |maz|
        if maz.tipo == tip
          array<< maz
        end
      end
      return array
    end
    
  end
  
  prueba = PruebaQytetet.new
  prueba.main
end
